sklearn/neighbors/_classification.py

```diff
@@ -69,6 +69,12 @@ class KNeighborsClassifier(KNeighborsMixin, ClassifierMixin, NeighborsBase):
         minkowski, and with p=2 is equivalent to the standard Euclidean
         metric. For a list of available metrics, see the documentation of
         :class:`~sklearn.metrics.DistanceMetric`.
+        Note that for cosine similarity 'cosine', its distance
```

Comment:
There are two things to add here:

1. `metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS` are available as metrics..
2. `cosine` distance is being used

With that in mind, I suggest the following adjustment:

> Metrics listed in `metrics.pairwise.PAIRWISE_DISTANCE_FUNCTIONS` are also supported. For the `'cosine'` metric, the :func:`~sklearn.metrics.pairwise.cosine_distances` is used.

As the API docs for `metrics` is getting long, I think it is a little better to immediately connect `'cosine'` to `cosine_distance`. The link to `cosine_distance` and the `cosine_distance` docstring should make it clear that the metric is `1 - similarity`.